### ISSUE TITLE
Support EX1 Events for fbw event handling

### ISSUE THREAD

**Issue by frankkopp** [flybywiresim/aircraft#8325]
### Aircraft Version

Development

### Description

Implement handling of EX1 events in the fbw c++ code.

Currently on non-EX1 events (only one data parameter) are handled by the event processing code:
https://github.com/flybywiresim/aircraft/blob/7e1f8614c7b91e10e3bbd1730677b9002d3f4672/fbw-a32nx/src/wasm/fbw_a320/src/interface/SimConnectInterface.cpp#L1427

A handler for EX1 events needs to be added. Easy approach would be to continue to assume only events with one parameter are used in the fbw code then a simple wrapper would do. Implementing EX1 full might not be necessary as long as multiple parameters are not required for the fbw code.  

Example from the c++ framework PR:
https://github.com/flybywiresim/aircraft/blob/61ef63b1b88cfb8e990a877b6893281c0418516a/fbw-common/src/wasm/cpp-msfs-framework/MsfsHandler/DataManager.cpp#L289
https://github.com/flybywiresim/aircraft/blob/61ef63b1b88cfb8e990a877b6893281c0418516a/fbw-common/src/wasm/cpp-msfs-framework/MsfsHandler/DataManager.cpp#L329
https://github.com/flybywiresim/aircraft/blob/61ef63b1b88cfb8e990a877b6893281c0418516a/fbw-common/src/wasm/cpp-msfs-framework/MsfsHandler/DataManager.cpp#L338

### References (optional)

https://docs.flightsimulator.com/flighting/html/Programming_Tools/SimConnect/API_Reference/Events_And_Data/SimConnect_TransmitClientEvent.htm

https://docs.flightsimulator.com/flighting/html/Programming_Tools/SimConnect/API_Reference/Events_And_Data/SimConnect_TransmitClientEvent_EX1.htm

### Discord Username (optional)

cdr_maverick

**Comment by tracernz** [comment]
Why do we want the EX1 events if we don't care about the extra params?

**Comment by frankkopp** [comment]
> Why do we want the EX1 events if we don't care about the extra params? Won't do any harm to use EX1 instead of the older interface of course.

Good point. Check the discussion here: https://discord.com/channels/738864299392630914/785976111875751956/1182012107546447945

Seems to make it easier for 3rd parties using the events. I tend to agree and I'm happy to look into it.

**Comment by tracernz** [comment]
Might be worth reporting that on devsupport because I don't think events should only be delivered to one of them.

**Comment by tracernz** [comment]
> I think this is intended as it wouldn't make sense to always fire both. Esp. for the EX1 version using more than 1 parameter which would not work with the non-EX version anyway.

If you do KeyEventManager.triggerKeyEvent in JS it doesn't get delivered to both? I'm pretty sure it does and this does seem like a bug to me. Generally you're supposed to use the _EX1 APIs instead of the original ones, not in parallel.

**Comment by tracernz** [comment]
After some research, they did not preserve backwards compatibility and the _EX1 event is orthogonal to the original. Older clients simply no longer receive events that would be routed to _EX1. So your proposed patch will be fine 😁 .

**Comment by frankkopp** [comment]
> After some research, they did not preserve backwards compatibility and the _EX1 event is orthogonal to the original. Older clients simply no longer receive events that would be routed to _EX1. So your proposed patch will be fine 😁 .

Thanks for testing it - it was really important to clarify this.

**Comment by HansBilliet** [comment]
Maybe not relevant any longer, because you guys seem to have done the implementation already. But it definitely makes sense to implement EX1 next to the the non_EX1 version. I'm working on a tool that also allows registering these Custom Events. I don't want to make a method that needs to have a parameter like "use EX1" or "use non_EX1". So I always want to use the "EX1"-version, which makes it much more simple for the user as well.

Question: Any idea when this would be available in the Development version? I'm on the latest one today, which is 7e1F861.
